src/angularComponent/ngcOmniboxController.js

```diff
@@ -125,16 +162,66 @@ export default class NgcOmniboxController {
     } else if (keyCode === KEY.DOWN) {
       this.highlightNext();
     } else if (keyCode === KEY.ESC) {
-      this.highlightedIndex = -1;
+      this.highlightNone();
+    } else if (isSelectKey(keyCode)) {
+      const selection = this._suggestionsUiList[this.highlightedIndex];
+      selection && this._selectItem(selection.data);
     }
   }
 
   _updateSuggestions() {
-    this._suggestionItems.length = 0;
-    this.highlightedIndex = -1;
-
-    this.source({query: this.ngModel}).then((suggestions) => {
-      this.suggestions = suggestions;
+    this._suggestionsUiList.length = 0;
+    this.highlightNone();
+
+    this.source({query: this.query}).then((suggestions) => {
+      if (suggestions) {
+        this.suggestions = Array.prototype.slice.apply(suggestions);
+        this._buildSuggestionsUiList();
+      } else {
+        this.suggestions = suggestions;
+      }
     });
   }
+
+  /**
+   * Builds a flat list of items out of the suggestions for easier keyboard navigation.
+   *
+   * @private
+   */
+  _buildSuggestionsUiList() {
```

Comment:
It seems like this needs to happen every time `suggestions` is changed. In your unit tests and the code above, you need to remember to call this manually. Why not just use a setter?

```
set suggestions(newSuggestions) {
  this._suggestions = newSuggestions;
  this._suggestionsUiList = flatten(newSuggestions);
}
```